File: docs/code_snippet/java/setupAppPayment1.java
// create your own app plan
AppPayment premiumPlan = new AppPayment(/* activity context */ this,/* amount */ 19.99, "Premium", MainActivity.class);

try {
    Chapa chapa = Chapa.getInstance();
    // simply call pay method without callback
     chapa.pay(this, premiumPlan);
    // or with callback
    // chapa.pay(this, premiumPlan, new ChapaPaymentCallback<AppPayment>() {
    //     @Override
    //     public void onSuccess(@NonNull String tx_ref, @NonNull AppPayment paymentType) {
    //         // TODO your code here
    //     }
    
    //     @Override
    //     public void onError(@NonNull ChapaError chapaError) {
    //         // TODO your code here
    //     }

    //     @Override
    //     public void onCancel() {
    //         // TODO your code here
    //     }
    // });
    
} catch (ChapaError e) {
    // CHAPA_NOT_INITIALIZED error
}

File: docs/code_snippet/kotlin/setupAppPayment1.kt
// create your own app plan
val premiumPlan = AppPayment(/* activity context */ this,/* amount */ 19.99, "Premium", MainActivity::class.java)
val chapa = Chapa.getInstance()
// simply call pay method without callback
chapa.pay(this,premiumPlan)
// or with callback
//chapa.pay(this,premiumPlan,object : ChapaPaymentCallback<AppPayment>{
//    override fun onSuccess(tx_ref: String, paymentType: AppPayment) {
//        // TODO your code here
//    }

//    override fun onError(chapaError: ChapaError) {
//        // TODO your code here
//    }

//    override fun onCancel() {
//        // TODO your code here
//    }
//})